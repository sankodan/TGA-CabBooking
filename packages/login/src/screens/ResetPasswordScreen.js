import React, { useState } from 'react'
import { Background, BackButton, TextInput, Button, Header, Logo, Toast } from '@sankodan/com.cabbooking.components'
import { emailValidator, sendEmailWithPassword } from '@sankodan/com.cabbooking.core'

export default function ResetPasswordScreen ({ navigation }) {
  const [email, setEmail] = useState({ value: '', error: '' })
  const [loading, setLoading] = useState(false)
  const [toast, setToast] = useState({ value: '', type: '' })

  const sendResetPasswordEmail = async () => {
    const emailError = emailValidator(email.value)
    if (emailError) {
      setEmail({ ...email, error: emailError })
      return
    }
    setLoading(true)
    const response = await sendEmailWithPassword(email.value)
    console.log("sendEmailWithPassword",response)
    if (response.error) {
      setToast({ type: 'error', message: response.error })
    } else {
      setToast({
        type: 'success',
        message: 'Email with password has been sent.'
      })
    }
    setLoading(false)
  }

  return (
    <Background>
      <BackButton goBack={navigation.goBack} />
      <Logo />
      <Header>Restore Password</Header>
      <TextInput
        testID = 'emailAddress'
        label="E-mail address"
        returnKeyType="done"
        value={email.value}
        onChangeText={(text) => setEmail({ value: text, error: '' })}
        error={!!email.error}
        errorText={email.error}
        autoCapitalize="none"
        autoCompleteType="email"
        textContentType="emailAddress"
        keyboardType="email-address"
        description="You will receive email with password reset link."
      />
      <Button
        testID = "resetPasswordEmailButton"
        loading={loading}
        mode="contained"
        onPress={sendResetPasswordEmail}
        style={{ marginTop: 16 }}
      >
        Send Instructions
      </Button>
      <Toast {...toast} testID="onDismiss" onDismiss={() => setToast({ value: '', type: '' })} />
    </Background>
  )
}
