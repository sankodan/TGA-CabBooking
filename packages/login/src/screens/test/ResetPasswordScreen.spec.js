import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react-native'
import ResetPasswordScreen from '../ResetPasswordScreen'
import { shallow } from 'enzyme'

describe('ResetPasswordScreen', () => {
  const navigate = jest.fn()
  const popToTop = jest.fn()
  const push = jest.fn()
  const navigation = {
    navigate,
    goBack: jest.fn(),
    popToTop,
    push
  }
  describe('clicking resetPasswordEmailButton', () => {
    let toJSONObject
    beforeEach(() => {
      const { toJSON } = render(<ResetPasswordScreen navigation={navigation} />)
      toJSONObject = toJSON
      fireEvent.changeText(screen.getByTestId('emailAddress'), 'test@test.com')
      fireEvent.press(screen.getByTestId('resetPasswordEmailButton'))
    })

    it('ResetPasswordScreen snapshot', () => {
      expect(toJSONObject()).toMatchSnapshot()
    })

    it('clears the message field', () => {
      expect(screen.getByTestId('emailAddress')).toHaveProp('value', 'test@test.com')
    })
  })

  describe('clicking resetPasswordEmailButton', () => {
    let toJSONObject
    beforeEach(() => {
      const { toJSON } = render(<ResetPasswordScreen navigation={navigation} />)
      toJSONObject = toJSON
      fireEvent.changeText(screen.getByTestId('emailAddress'), 'test')
      fireEvent.press(screen.getByTestId('resetPasswordEmailButton'))
    })

    it('ResetPasswordScreen snapshot', () => {
      expect(toJSONObject()).toMatchSnapshot()
    })

    it('clears the message field', () => {
      expect(screen.getByTestId('emailAddress')).toHaveProp('value', 'test')
    })
  })
})
