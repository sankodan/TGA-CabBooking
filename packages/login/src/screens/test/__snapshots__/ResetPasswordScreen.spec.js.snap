// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordScreen clicking resetPasswordEmailButton ResetPasswordScreen snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 4,
          "opacity": 1,
          "position": "absolute",
          "top": 60,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/components/assets/arrow_back.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../packages/components/assets/logo.png",
        }
      }
      style={
        Object {
          "height": 110,
          "marginBottom": 8,
          "width": 110,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#121330",
            "fontSize": 21,
            "fontWeight": "bold",
            "paddingVertical": 12,
          },
        ]
      }
    >
      Restore Password
    </Text>
    <View
      style={
        Object {
          "marginVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                E-mail address
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                E-mail address
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                E-mail address
              </Text>
            </View>
            <TextInput
              autoCapitalize="none"
              autoCompleteType="email"
              editable={true}
              keyboardType="email-address"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              returnKeyType="done"
              selectionColor="#121330"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="emailAddress"
              textContentType="emailAddress"
              underlineColorAndroid="transparent"
              value="test@test.com"
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#414757",
            "fontSize": 13,
            "paddingTop": 8,
          }
        }
      >
        You will receive email with password reset link.
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginTop": 16,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="resetPasswordEmailButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Send Instructions
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "top": 180,
          "width": "100%",
        }
      }
      testID="onDismiss"
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "bottom": 0,
              "position": "absolute",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityLiveRegion="polite"
          collapsable={false}
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00B386",
              "borderRadius": 4,
              "elevation": 6,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 8,
              "opacity": 0,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 6,
              "transform": Array [
                Object {
                  "scale": 0.9,
                },
              ],
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "flex": 1,
                    "flexWrap": "wrap",
                    "marginLeft": 16,
                    "marginVertical": 14,
                  },
                  Object {
                    "color": "#ffffff",
                    "marginRight": 16,
                  },
                ],
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "500",
                }
              }
            >
              Email with password has been sent.
            </Text>
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</View>
`;

exports[`ResetPasswordScreen clicking resetPasswordEmailButton ResetPasswordScreen snapshot 2`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 4,
          "opacity": 1,
          "position": "absolute",
          "top": 60,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/components/assets/arrow_back.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../packages/components/assets/logo.png",
        }
      }
      style={
        Object {
          "height": 110,
          "marginBottom": 8,
          "width": 110,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#121330",
            "fontSize": 21,
            "fontWeight": "bold",
            "paddingVertical": 12,
          },
        ]
      }
    >
      Restore Password
    </Text>
    <View
      style={
        Object {
          "marginVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#B00020",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                E-mail address
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#B00020",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                E-mail address
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "#B00020",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 1,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                E-mail address
              </Text>
            </View>
            <TextInput
              autoCapitalize="none"
              autoCompleteType="email"
              editable={true}
              keyboardType="email-address"
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              returnKeyType="done"
              selectionColor="#121330"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontVariant": undefined,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="emailAddress"
              textContentType="emailAddress"
              underlineColorAndroid="transparent"
              value="test"
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#f13a59",
            "fontSize": 13,
            "paddingTop": 8,
          }
        }
      >
        Ooops! We need a valid email address.
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginTop": 16,
          "marginVertical": 10,
          "minWidth": 64,
          "paddingVertical": 2,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
        testID="resetPasswordEmailButton"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "lineHeight": 26,
                  },
                ],
              ]
            }
          >
            Send Instructions
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "top": 180,
          "width": "100%",
        }
      }
      testID="onDismiss"
    />
  </View>
</View>
`;
